src/joystick/JoystickManager.cpp

```diff
@@ -0,0 +1,111 @@
+#include <iostream>
+#include <chrono>
+
+#include <SDL.h>
+#include <SDL_joystick.h>
+
+#include "JoystickManager.h"
+
+using namespace std::chrono;
+
+JoystickManager::JoystickManager() {
+    std::cout << "[JoystickManager] New JoystickManager" << std::endl;
+    init();
+    loop();
+}
+
+/**
+ * Inits SDL and JoystickHandler-map
+ */
+void JoystickManager::init() {
+    std::cout << "[JoystickManager][init] Initializing.." << std::endl;
+    if(SDL_InitSubSystem(SDL_INIT_JOYSTICK))
+        std::cout << "SDL_INIT_JOYSTICK failed : " << SDL_GetError() << std::endl;
+    SDL_JoystickEventState(SDL_ENABLE);
+}
+
+/**
+ * Handles events, handles and ticks JoystickHandlers
+ */
+void JoystickManager::loop() {
```

Comment:
function exceeds recommended 20 lines. can be easily split up in smaller functions